tasks/sync.js

```diff
@@ -147,7 +139,7 @@ module.exports = function (grunt) {
 
   function processPair (justPretend, failOnError, logger, comparatorFactory, src, dest, copyOptions) {
     // stat destination file
-    return promise.all([fs.stat(src), fs.stat(dest)]).then(function (result) {
+    return Promise.all([fs.stat(src), fs.stat(dest)]).then(function (result) {
```

Comment:
`fs.stat` doesn't exist in `fs-extra`, this method throws because no callback is provided and it doesn't return a promise. Top of the callstack is https://github.com/nodejs/node/blob/ebf5b58bec6954ebeed1c302cf31ff55494a0c93/lib/fs.js#L795

Reply:
it is due to this change in node 10 https://github.com/nodejs/node/pull/20220/files

Reply:
`require('fs').promises` is only available in Node 10. It doesn't really seem fair for this plugin to only support Node 10+.

Regardless, `fs-extra` is working just fine for me. `fs.stat()` does return a promise. I just tried this code on my machine (Node 10.5.0) and it worked:

```js
const fs = require('fs-extra');
fs.stat('./index.js').then(stat => console.log(stat));
```

Which version of Node are you running @benjamin-demarteau?

Reply:
Alright, ignore me, I found out that some dependency required fs-extra 0.30 which caused the resolver to use that for grunt-sync. That version is obviously very broken, updating it works fine.